### ISSUE TITLE
Fix 'test_feeds/test_invalid_syntax' tests for canonical, rehat, and MSU in 4.2

### ISSUE THREAD

**Issue by damarisg** [wazuh/wazuh-qa#1547]
### Issue information

| Related issue |
|--|
| Closes https://github.com/wazuh/wazuh-qa/issues/1531|

When it comes to starting with the test fix, there is some information that can help you:
```
Module: Vulnerability Detector
Support Team:  "binary beasts"
Target: Manager   
OS: Linux
```
I add some information obtained while researched the logs that failed.

## Case 1:

|  Type | Description  |
|--|--|
|Test Path | test_feeds/canonical/test_invalid_syntax_canonical_feed
| Consistent | yes
| Test Execution | 3/3 executions failed
| Cases Fails | 3
| Summary | On `test_invalid_syntax_canonical_feed` method when it called to `check_failure_when_importing_feed`.  We need to see if there is a problem to import a file or if it's a Parser error in log event.
| Error message |  "Could not find the log event: ERROR: \\(\\d+\\): CVE database could not be updated."

## Case 2:

|  Type | Description  |
|--|--|
|Test Path | test_feeds/redhat/test_invalid_syntax_redhat_feed.  
| Consistent | no
| Test Execution | 2/3 executions failed
| Cases Fails | 6
| Summary | It seems to be like we don't detect the event correctly. 
| Error message |  "Could not find the log event: ERROR: \\(\\d+\\): CVE database could not be updated."

## Case 3:

|  Type | Description  |
|--|--|
|Test Path | test_feeds/msu/test_invalid_syntax_msu_feed
| Consistent | yes
| Test Execution | 3/3 executions failed
| Cases Fails | 46
| Summary | On `test_invalid_syntax_msu_feed` method when called `check_failure_when_importing_feed`. The problem could be a Parser error when check in the log or a problem when file is imported.
| Error message |  "Could not find the log event: ERROR: \\(\\d+\\): The .* feed couldn't be parsed from .* file"

In order to finish this issue the following tasks should be fulfilled:
- [ ] Research of fails.
- [ ] Apply Fix 
- [ ] Full Green/ Full Yellow  in `test_feeds/canonical/test_invalid_syntax_canonical_feed` for 3 times.
- [ ] Full Green/ Full Yellow  in `test_feeds/redhat/test_invalid_syntax_redhat_feed.` for 3 times.
- [ ] Full Green/ Full Yellow in ` test_feeds/msu/test_invalid_syntax_msu_feed` for 3 times. 
- [ ] Documentation of any important change done for these tests or the used tools.

**Comment by mdengra** [comment]
### 2021-07-16

Test results with the modules `sca`, `syscollector` and `rootcheck` modules disabled in the `ossec.conf`: 

### Case 1
| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[test_invalid_syntax_canonical_feed_local_dm_r1.log](https://github.com/wazuh/wazuh-qa/files/6830361/test_invalid_syntax_canonical_feed_local_dm_r1.log)| 2021-07-16 | Miguel | :yellow_circle: 
|[test_invalid_syntax_canonical_feed_local_dm_r2.log](https://github.com/wazuh/wazuh-qa/files/6830362/test_invalid_syntax_canonical_feed_local_dm_r2.log)| 2021-07-16 | Miguel | :yellow_circle: 
|[test_invalid_syntax_canonical_feed_local_dm_r3.log](https://github.com/wazuh/wazuh-qa/files/6830363/test_invalid_syntax_canonical_feed_local_dm_r3.log)| 2021-07-16 | Miguel | :yellow_circle: 

### Case 2
| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[test_invalid_syntax_redhat_feed_local_dm_r1.log](https://github.com/wazuh/wazuh-qa/files/6830364/test_invalid_syntax_redhat_feed_local_dm_r1.log)| 2021-07-16 | Miguel | :red_circle: 
|[test_invalid_syntax_redhat_feed_local_dm_r2.log](https://github.com/wazuh/wazuh-qa/files/6830365/test_invalid_syntax_redhat_feed_local_dm_r2.log)| 2021-07-16 | Miguel | :red_circle: 
|[test_invalid_syntax_redhat_feed_local_dm_r3.log](https://github.com/wazuh/wazuh-qa/files/6830366/test_invalid_syntax_redhat_feed_local_dm_r3.log)| 2021-07-16 | Miguel | :red_circle: 

Note that the same use case fails in all runs:
```
FAILED test_vulnerability_detector/test_feeds/redhat/test_invalid_syntax_redhat_feed.py::test_extra_fields_redhat_feed[REDHAT_configuration-Delete '<']
```

### Case 3
| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[test_invalid_syntax_msu_feed_local_dm_r1.log](https://github.com/wazuh/wazuh-qa/files/6830370/test_invalid_syntax_msu_feed_local_dm_r1.log)| 2021-07-16 | Miguel | :yellow_circle: 
|[test_invalid_syntax_msu_feed_local_dm_r2.log](https://github.com/wazuh/wazuh-qa/files/6830372/test_invalid_syntax_msu_feed_local_dm_r2.log)| 2021-07-16 | Miguel | :yellow_circle: 
|[test_invalid_syntax_msu_feed_local_dm_r3.log](https://github.com/wazuh/wazuh-qa/files/6830377/test_invalid_syntax_msu_feed_local_dm_r3.log)| 2021-07-16 | Miguel | :yellow_circle:

**Comment by mdengra** [comment]
### 2021-07-19

Used Wazuh-QA branch: `1531-full-yellow-vuln-det`
Test results with the modules `sca`, `syscollector` and `rootcheck` modules disabled in the `ossec.conf`: 

### Case 1
| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[test_invalid_syntax_canonical_feed_local_dm_r1.log](https://github.com/wazuh/wazuh-qa/files/6840722/test_invalid_syntax_canonical_feed_local_dm_r1.log)| 2021-07-19 | Miguel | :yellow_circle: 
|[test_invalid_syntax_canonical_feed_local_dm_r2.log](https://github.com/wazuh/wazuh-qa/files/6840725/test_invalid_syntax_canonical_feed_local_dm_r2.log)| 2021-07-19 | Miguel | :yellow_circle: 
|[test_invalid_syntax_canonical_feed_local_dm_r3.log](https://github.com/wazuh/wazuh-qa/files/6840735/test_invalid_syntax_canonical_feed_local_dm_r3.log)| 2021-07-19 | Miguel | :yellow_circle: 


### Case 2
| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[test_invalid_syntax_redhat_feed_local_dm_r1.log](https://github.com/wazuh/wazuh-qa/files/6840747/test_invalid_syntax_redhat_feed_local_dm_r1.log)| 2021-07-19 | Miguel | :yellow_circle: 
|[test_invalid_syntax_redhat_feed_local_dm_r2.log](https://github.com/wazuh/wazuh-qa/files/6840749/test_invalid_syntax_redhat_feed_local_dm_r2.log)| 2021-07-19 | Miguel | :yellow_circle: 
|[test_invalid_syntax_redhat_feed_local_dm_r3.log](https://github.com/wazuh/wazuh-qa/files/6840757/test_invalid_syntax_redhat_feed_local_dm_r3.log)| 2021-07-19 | Miguel | :yellow_circle: 


### Case 3
| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[test_invalid_syntax_msu_feed_local_dm_r1.log](https://github.com/wazuh/wazuh-qa/files/6840773/test_invalid_syntax_msu_feed_local_dm_r1.log)| 2021-07-19 | Miguel | :yellow_circle: 
|[test_invalid_syntax_msu_feed_local_dm_r2.log](https://github.com/wazuh/wazuh-qa/files/6840774/test_invalid_syntax_msu_feed_local_dm_r2.log)| 2021-07-19 | Miguel | :yellow_circle: 
|[test_invalid_syntax_msu_feed_local_dm_r3.log](https://github.com/wazuh/wazuh-qa/files/6840781/test_invalid_syntax_msu_feed_local_dm_r3.log)| 2021-07-19 | Miguel | :yellow_circle:

**Comment by mdengra** [comment]
### 2021-07-22

Used Wazuh-QA branch: `1531-full-yellow-vuln-det`
Test results with the default settings in the `ossec.conf`: 

### Case 1
| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[test_invalid_syntax_canonical_feed_local_r1.log](https://github.com/wazuh/wazuh-qa/files/6861210/test_invalid_syntax_canonical_feed_local_r1.log)| 2021-07-22 | Miguel | :yellow_circle: 
|[test_invalid_syntax_canonical_feed_local_r2.log](https://github.com/wazuh/wazuh-qa/files/6861211/test_invalid_syntax_canonical_feed_local_r2.log)| 2021-07-22 | Miguel | :yellow_circle: 
|[test_invalid_syntax_canonical_feed_local_r3.log](https://github.com/wazuh/wazuh-qa/files/6861213/test_invalid_syntax_canonical_feed_local_r3.log)| 2021-07-22 | Miguel | :yellow_circle: 


### Case 2
| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[test_invalid_syntax_redhat_feed_local_r1.log](https://github.com/wazuh/wazuh-qa/files/6861446/test_invalid_syntax_redhat_feed_local_r1.log)| 2021-07-22 | Miguel | :red_circle: 
|[test_invalid_syntax_redhat_feed_local_r2.log](https://github.com/wazuh/wazuh-qa/files/6861449/test_invalid_syntax_redhat_feed_local_r2.log)| 2021-07-22 | Miguel | :red_circle: 
|[test_invalid_syntax_redhat_feed_local_r3.log](https://github.com/wazuh/wazuh-qa/files/6861450/test_invalid_syntax_redhat_feed_local_r3.log)| 2021-07-22 | Miguel | :yellow_circle: 

Note that the same use case (first case) fails in the two first runs:
```
FAILED test_vulnerability_detector/test_feeds/redhat/test_invalid_syntax_redhat_feed.py::test_extra_fields_redhat_feed[REDHAT_configuration-Delete '<']
```

### Case 3
| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[test_invalid_syntax_msu_feed_local_r1.log](https://github.com/wazuh/wazuh-qa/files/6861441/test_invalid_syntax_msu_feed_local_r1.log)| 2021-07-22 | Miguel | :yellow_circle: 
|[test_invalid_syntax_msu_feed_local_r2.log](https://github.com/wazuh/wazuh-qa/files/6861442/test_invalid_syntax_msu_feed_local_r2.log)| 2021-07-22 | Miguel | :yellow_circle: 
|[test_invalid_syntax_msu_feed_local_r3.log](https://github.com/wazuh/wazuh-qa/files/6861444/test_invalid_syntax_msu_feed_local_r3.log)| 2021-07-22 | Miguel | :yellow_circle:

**Comment by mdengra** [comment]
The problem is related to the issue: https://github.com/wazuh/wazuh-qa/issues/1602, in which `FileMonitor` does not detect a message written to the log file under certain scenarios.

In the `test_invalid_syntax_redhat_feed.py` test, even disabling all possible modules before launching the test does not prevent it from falling in the first use case. 

The cause of the failure is the large number of messages generated by enabling the internal option `wazuh_modules.debug=2`, which is required by multiple `vulnerability detector` tests. When this option is enabled and the manager is restarted, the log file takes up 1MB, this added to the messages generated by the test itself makes `FileMonitor` unable to process them, causing the error.

**Comment by damarisg** [comment]
Closed by #1630 

Test Results after to merge:

| Test Executions | Date  |Test |  By  | Status | 
|--|--|--|--|--|
|[YellowInvalidSyntaxCanonicalFeed.log](https://github.com/wazuh/wazuh-qa/files/6870234/YellowInvalidSyntaxCanonicalFeed.log) | 2021-07-23 | test_feeds/canonical/test_invalid_syntax_canonical_feed |Seyla| 🟡 
|[YellowInvalidSyntaxCanonicalFeed2.log](https://github.com/wazuh/wazuh-qa/files/6870237/YellowInvalidSyntaxCanonicalFeed2.log)| 2021-07-23 | test_feeds/canonical/test_invalid_syntax_canonical_feed |Seyla| 🟡 
|[YellowInvalidSyntaxCanonicalFeed3.log](https://github.com/wazuh/wazuh-qa/files/6870239/YellowInvalidSyntaxCanonicalFeed3.log)| 2021-07-23 | test_feeds/canonical/test_invalid_syntax_canonical_feed |Seyla| 🟡 
|[YellowInvalidSyntaxRedhatFeed.log](https://github.com/wazuh/wazuh-qa/files/6870348/YellowInvalidSyntaxRedhatFeed.log)| 2021-07-23 | test_feeds/redhat/test_invalid_syntax_redhat_feed |Seyla|🟡 
|[YellowInvalidSyntaxRedhatFeed2.log](https://github.com/wazuh/wazuh-qa/files/6870349/YellowInvalidSyntaxRedhatFeed2.log)| 2021-07-23 | test_feeds/redhat/test_invalid_syntax_redhat_feed |Seyla|🟡 
|[YellowInvalidSyntaxRedhatFeed3.log](https://github.com/wazuh/wazuh-qa/files/6870350/YellowInvalidSyntaxRedhatFeed3.log)| 2021-07-23 | test_feeds/redhat/test_invalid_syntax_redhat_feed |Seyla|🟡 
|[YellowInvalidSyntaxMsuFeed.log](https://github.com/wazuh/wazuh-qa/files/6870354/YellowInvalidSyntaxMsuFeed.log)| 2021-07-23 | test_feeds/msu/test_invalid_syntax_msu_feed |Seyla|🟡 
|[YellowInvalidSyntaxMsuFeed2.log](https://github.com/wazuh/wazuh-qa/files/6870346/YellowInvalidSyntaxMsuFeed2.log)| 2021-07-23 | test_feeds/msu/test_invalid_syntax_msu_feed |Seyla|🟡 
|[YellowInvalidSyntaxMsuFeed3.log](https://github.com/wazuh/wazuh-qa/files/6870347/YellowInvalidSyntaxMsuFeed3.log)| 2021-07-23 | test_feeds/msu/test_invalid_syntax_msu_feed |Seyla|🟡
